solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/PerReplicaStatesFetcher.java

```diff
@@ -28,21 +29,26 @@
 
 public class PerReplicaStatesFetcher {
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private static final ConcurrentHashMap<String, PerReplicaStates> fetchedStates =
```

Comment:
`static` field storing all PRS ? This is a red flag. We shouldn't keep unbounded caches of data